sdk/storage/src/common/storage_url_builder.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "common/storage_url_builder.hpp"
+
+#include <locale>
+#include <vector>
+#include <algorithm>
+#include <cctype>
+
+namespace Azure { namespace Storage {
+
+  UrlBuilder::UrlBuilder(const std::string& url)
+  {
+    std::string::const_iterator pos = url.begin();
+
+    const std::string scheme_end = "://";
+    auto scheme_i = url.find(scheme_end);
+    if (scheme_i != std::string::npos)
+    {
+      std::transform(url.begin(), url.begin() + scheme_i, std::back_inserter(m_scheme), [](char c) {
+        return static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
+      });
+      pos = url.begin() + scheme_i + scheme_end.length();
+    }
+
+    auto host_ite = std::find_if(pos, url.end(), [](char c) { return c == '/' || c == '?' || c == ':'; });
+    m_host = std::string(pos, host_ite);
+    pos = host_ite;
+
+    if (pos != url.end() && *pos == ':')
+    {
+      auto port_ite = std::find_if_not(pos + 1, url.end(), [](char c) { return std::isdigit(static_cast<unsigned char>(c)); });
+      m_port = std::stoi(std::string(pos + 1, port_ite));
+      pos = port_ite;
+    }
+
+    if (pos != url.end() && *pos == '/')
+    {
+      auto path_ite = std::find(pos + 1, url.end(), '?');
```

Comment:
>path_ite [](start = 11, length = 8)

Minor: as talked offline, naming of the parameters should not be using snake case.